core/src/main/java/com/novoda/noplayer/internal/exoplayer/ExoPlayerTwoImpl.java

```diff
@@ -272,6 +273,18 @@ public void detach(PlayerView playerView) {
         this.playerView = null;
     }
 
+    @Override
+    public void attach(AdvertView advertView) {
+        listenersHolder.addAdvertListener(advertView.getAdvertListener());
```

Comment:
Attached to the collection of advert listeners so that the `AdvertView` and whatever other listeners the client attaches are notified of advert events.